src/wp-includes/block-template-utils.php

```diff
@@ -507,6 +507,64 @@ function _remove_theme_attribute_from_template_part_block( &$block ) {
 	}
 }
 
+/**
+ * Gets the contents for the given template file path, using a transient cache.
+ *
+ * @since 6.4.0
+ * @access private
+ *
+ * @param string $template_file_path Absolute path to a theme template file.
+ * @return string The template file contents.
+ */
+function _get_block_template_file_content( $template_file_path ) {
+	$theme = wp_get_theme();
+
+	$template_file_path = wp_normalize_path( $template_file_path );
+	$theme_dir          = wp_normalize_path( $theme->get_stylesheet_directory() ) . '/';
+
+	if ( str_starts_with( $template_file_path, $theme_dir ) ) {
+		$relative_path = substr( $template_file_path, strlen( $theme_dir ) );
+	} elseif ( $theme->parent() ) {
+		$theme     = $theme->parent();
+		$theme_dir = wp_normalize_path( $theme->get_stylesheet_directory() ) . '/';
+		if ( str_starts_with( $template_file_path, $theme_dir ) ) {
+			$relative_path = substr( $template_file_path, strlen( $theme_dir ) );
+		}
+	}
+
+	// Bypass cache if the file is not within the theme directory.
+	if ( ! isset( $relative_path ) ) {
+		return file_get_contents( $template_file_path );
+	}
+
+	// Bypass cache while developing a theme.
+	$can_use_cache = ! wp_is_development_mode( 'theme' );
+	if ( ! $can_use_cache ) {
+		return file_get_contents( $template_file_path );
+	}
+
+	// Check theme template cache first (if cached version matches the current theme version).
+	$template_data = get_transient( 'wp_theme_template_contents_' . $theme->get_stylesheet() );
+	if ( is_array( $template_data ) && $template_data['version'] === $theme->get( 'Version' ) ) {
+		if ( isset( $template_data['template_content'][ $relative_path ] ) ) {
+			return $template_data['template_content'][ $relative_path ];
+		}
```

Comment:
```suggestion
	if ( isset( $template_data['template_content'][ $relative_path ] ) && $template_data['version'] === $theme->get( 'Version' ) ) {
		return $template_data['template_content'][ $relative_path ];
```
Not a blocker, just curious what you think 🙂

Reply:
@costdev Doing this would change the logic in a way that would break it: In the outer clause, we should only check whether the _overall_ cache has valid data, because the `else` otherwise reinitializes the entire cache object.

This cache is used for different instances of this function, i.e. different `$relative_path`s, so if the specific `$relative_path` we're looking for is missing and the cache is intact, we should add it to the cache, not reinitialize the entire cache.